### ISSUE TITLE
Fix the reference feature not working consistently

### ISSUE THREAD

**Issue by jimleroyer** [cds-snc/notification-planning#398]
# Description

The [reference feature](https://documentation.notification.canada.ca/en/status.html#get-the-status-of-multiple-messages) is not getting always through, or it is not carried in a consistent manner through the data pipeline.

Steve wrote:
> oh my beautiful pony - users can supply a reference field when POSTing to the api, 
> but that doesn't get put into the reference field of the notification, 
> that's something else. Gets put into the client_reference field instead

#  Steps to reproduce the behavior:

1. Send a notification through either the bulk API or the atomic api [which contains the `reference` field](https://documentation.notification.canada.ca/en/send.html#arguments).
2. Retrieve the messages containing that reference through the API to retrieve multiple messages [as described in the documentation](https://documentation.notification.canada.ca/en/status.html#get-the-status-of-multiple-messages).

# Expected behavior

The notifications should pass, persist and be retrieved by/with the custom reference passed by the user.

**Additional context**

This is in the context of testing the bulk API, as the reference is a feature that makes it easy for us to sequentially tag messages and retrieve these back, to see if any are missing.

**Comment by sastels** [comment]
Verified that `reference` is getting through using redis batch saving of atomic emails (ie POST to `/v2/notifications/email`).

Verified that it doesn't work with posts to `/v2/notifications/bulk`. Looking at the code, we don't pull out a `reference` from anywhere.

Note that we don't mention `reference` in the `bulk` docs (https://documentation.notification.canada.ca/en/send.html#sending-notifications-in-bulk). Having said that, it would be useful if we supported this. For example, if we could POST rows of the form
```
{
  "name": "bulk name",
  "template_id": "9544dc32-9d23-4bdd-a8d8-81b4ec9e0485",
  "rows": [
    ["email address", "reference", "var"],
    ["stephen.astels@cds-snc.ca", "r1", "test 1"],
    ["stephen.astels@cds-snc.ca", "r2", "test 2"]
  ]
}
```
